internal/xds/runner/runner.go

```diff
@@ -253,14 +256,29 @@ func registerServer(srv serverv3.Server, g *grpc.Server) {
 	runtimev3.RegisterRuntimeDiscoveryServiceServer(g, srv)
 }
 
-func (r *Runner) translateFromSubscription(sub <-chan watchable.Snapshot[string, *ir.Xds]) {
+func (r *Runner) translateFromSubscription(sub <-chan watchable.Snapshot[string, *message.XdsIRWithContext]) {
 	// Subscribe to resources
 	message.HandleSubscription(message.Metadata{Runner: r.Name(), Message: message.XDSIRMessageName}, sub,
-		func(update message.Update[string, *ir.Xds], errChan chan error) {
+		func(update message.Update[string, *message.XdsIRWithContext], errChan chan error) {
 			r.Logger.Info("received an update")
+
+			parentCtx := context.Background()
+			if update.Value != nil && update.Value.Context != nil {
+				parentCtx = update.Value.Context
+			}
+
+			_, span := tracer.Start(parentCtx, "XdsRunner.subscribeAndTranslate")
```

Comment:
can we add another child span to ` r.cache.GenerateNewSnapshot` and use the trace id as the version inside it, so this id trickles into envoy proxy